integrations/git_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	git "gopkg.in/src-d/go-git.v4"
+)
+
+func TestClonePush_ViaHTTP_NoLogin(t *testing.T) {
+	prepareTestEnv(t)
+
+	s := http.Server{
+		Addr:    ":3000",
```

Comment:
Why not use a temporary Port? using `s.URL` to get the result?

Reply:
@bkcsoft It seems there is no `s.URL`.

Reply:
Here's how that's done: https://gist.github.com/xcsrz/538e291d12be6ee9a8c7#file-server-go-L13-L19